api/types/filters/parse.go

```diff
@@ -199,6 +200,36 @@ func (args Args) Match(field, source string) bool {
 	return false
 }
 
+// ParseBoolIfPresent overwrites the target with a boolean value of the key if it's present.
+// If the value cannot be interpreted as a boolean an error is returned.
+func (args Args) ParseBoolIfPresent(target *bool, key string) error {
+	fieldValues, ok := args.fields[key]
+
+	if !ok {
+		return nil
+	}
+
+	if len(fieldValues) == 0 {
```

Comment:
Maybe instead we ca error if `len(fieldValues) != 1` which would also simplifly the conflict logic.

Reply:
Oh, I forgot about this PR, thanks for reminding me. I'll try to update it once I have some spare time!

IIRC I did this this way specifically to not change the old behaviour of `filter.ExactMatch("dangling", "true") || filter.ExactMatch("dangling", "1") `, which doesn't error when you specify the filter twice with different syntax (true/1 or false/0) but with the same value. This is only the case for image prune and volume service list/prune though.

If we don't need to keep this behaviour then we can go with `len != 1`.